spec.md

```diff
@@ -388,13 +388,34 @@ both appear within the same JSON object.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
+### application-properties
+* Type: `Map`
+* Description: A list of "key-value" pairs providing information about the
+  event, which are used by an event-based application. The "key-value" pairs
+  MUST be in a flat structure. The value MUST be a string. There MUST be no
+  duplication of keys. One usage example: an application workflow can use a
+  "key-value" pair to correlate multiple events (from different event sources)
+  associated with a serverless application workflow.
+* Constraints:
+  * REQUIRED
```

Comment:
Leaving a question here that I also brought up in the call, so that it can be clarified in the text: If my application doesn't need to correlate events, what should go in this field, since it's required for the event?

Reply:
I don't think this should be a REQUIRED attribute since not all events need this - for example, we didn't use/need this in our demo at KubeCon so I'm not sure why it needs to be required.

Reply:
An event could be used by multiple applications. If one of the applications needs to correlate events and the information is not in the event, then that application can not work. Since the event producer and the serverless application or serverless platform are different entities, the event producer may not be able to predict how the applications or platforms will use it. To support those applications or platforms that need to correlate events, we should require the event producer to put the info in the event attribute

Reply:
The "required" is not from the event's perspective, but from the application/platform's perspective, which need to use the info for correlation purpose.

Reply:
not every application is going to be correlating events, and even when correlation is required, the source (and eventType) attribute will often be sufficient

Reply:
Mark: Yes not every application is going to correlate events. But if one application does it, the information MUST be there. Can we guarantee the information will be in the source attribute ?

Reply:
I remember we discussed this before. It is mandatory but can be empty

Reply:
If it can be empty, then I don't see the need to require an empty bag.